src/containers/FilterBarContainer.tsx

```diff
@@ -0,0 +1,50 @@
+import { connect } from "react-redux";
+import * as React from "react";
+import { Store, asyncActions, facetsActions, searchParameterActions } from "azsearchstore";
+import * as redux from "redux";
+import FilterBar from "../components/FilterBar";
+
+function getReturnType<RT>(expression: (...params: any[]) => RT): RT {
+    return {} as RT;
+}
+
+export interface OwnProps {
+}
+
+const mapDispatchToProps = (dispatch: redux.Dispatch<any>, ownProps: OwnProps) => {
+  return {
+    onClear: () => {
+      dispatch(facetsActions.clearFacetsSelections);
+      dispatch(searchParameterActions.setPage(1));
+      dispatch(asyncActions.fetchSearchResults);
+    }
+  };
+};
+
+function mapStateToProps(state: Store.SearchState, ownProps: OwnProps) {
+  return {
+    hasSelectedFacets: checkForAppliedFacets(state.facets.facets)
+  };
+}
+
+function checkForAppliedFacets(facets: { [key: string]: Store.Facet }) {
+  for (let facetKey in facets) {
+    let checkboxFacet = facets[facetKey] as Store.CheckboxFacet;
```

Comment:
This isn't a safe assumption is it? facet can also be of type RangeFacet. See an example of how I handle this in the [store for clearing facets](https://github.com/EvanBoyle/AzSearchStore/blob/master/src/reducers/facets.ts#L37). I haven't had a chance to incorperate it into the code yet, but this looks like a perfect candidate for[memoized selectors](http://redux.js.org/docs/recipes/ComputingDerivedData.html). Eventually it'll be necessary to rework all the containers to use selectors when appropriate for performance, but we can hold off for now while we're churning code.